src/transaction.rs

```diff
@@ -0,0 +1,229 @@
+use std::io::Error;
+
+use futures::{Poll, Stream};
+
+use {Config, Key, KvFuture, KvPair, Value};
+
+#[derive(Copy, Clone)]
+pub struct Timestamp(u64);
+
+impl Into<Timestamp> for u64 {
+    fn into(self) -> Timestamp {
+        Timestamp(self)
+    }
+}
+
+impl Timestamp {
+    pub fn timestamp(self) -> u64 {
+        self.0
+    }
+
+    pub fn physical(self) -> i64 {
+        (self.0 >> 16) as i64
+    }
+
+    pub fn logical(self) -> i64 {
+        (self.0 & 0xFFFF as u64) as i64
+    }
+}
+
+pub struct Scanner;
+
+impl Stream for Scanner {
+    type Item = KvPair;
+    type Error = Error;
+
+    fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
+        unimplemented!()
+    }
+}
+
+pub trait Retriever {
+    fn get<K>(&self, key: K) -> KvFuture<Value>
+    where
+        K: Into<Key>;
+
+    fn batch_get<I, K>(&self, keys: I) -> KvFuture<Vec<KvPair>>
+    where
+        I: IntoIterator<Item = K>,
+        K: Into<Key>;
+
+    fn seek<K>(&self, key: K) -> KvFuture<Scanner>
```

Comment:
when to set the scan range and limit?

Reply:
@siddontang I added set_limit and set_key_only to Scanner. It follows the naming convention of getter/setter. But it somehow looks strange, please take another look.

Reply:
I removed the set_limit and set_key_only from Scanner as key only is configured at Snapshot in Go client. And Go client only has batch size option and there is no actual limit on Scanner. I guess it was designed to let caller to decide when to stop polling result from Scanner.

Reply:
I'm a little worried about the design of Transactional API. There is no enough discussion on it's design and usage. Shell we remove it at this version or mark it as experimental to signal people that it might change in future.
@Hoverbear @siddontang